tst/EnergyPlus/unit/StandardRatings.unit.cc

```diff
@@ -0,0 +1,232 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus::StandardRatings unit tests
+
+// Google test headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <EnergyPlus/StandardRatings.hh>
+#include <EnergyPlus/CurveManager.hh>
+#include <EnergyPlus/DataGlobals.hh>
+#include <EnergyPlus/DXCoils.hh>
+#include <EnergyPlus/Psychrometrics.hh>
+
+using namespace EnergyPlus;
+using namespace EnergyPlus::StandardRatings;
+using namespace EnergyPlus::CurveManager;
+using namespace EnergyPlus::DataHVACGlobals;
+using namespace EnergyPlus::DXCoils;
+
+namespace EnergyPlus {
+
+	TEST_F( EnergyPlusFixture, SingleSpeedHeatingCoilCurveTest ) {
+		// Test that the standard ratings calculation with negative curve value
+
+		using CurveManager::Quadratic;
+		using CurveManager::Cubic;
+		using CurveManager::NumCurves;
+		using Psychrometrics::PsyRhoAirFnPbTdbW;
+		using DXCoils::DXCoil;
+		using StandardRatings::SingleSpeedDXHeatingCoilStandardRatings;
+
+
+		// Set up heating coil and curves.
+		int DXCoilNum;
+		NumDXCoils = 1;
+		DXCoilNum = 1;
+		DXCoil.allocate( NumDXCoils );
+		DXCoilNumericFields.allocate( 1 );
+		DXCoilOutletTemp.allocate( NumDXCoils );
+		DXCoilOutletHumRat.allocate( NumDXCoils );
+		DXCoilFanOpMode.allocate( NumDXCoils );
+		DXCoilPartLoadRatio.allocate( NumDXCoils );
+		DXCoilTotalHeating.allocate( NumDXCoils );
+		DXCoilHeatInletAirDBTemp.allocate( NumDXCoils );
+		DXCoilHeatInletAirWBTemp.allocate( NumDXCoils );
+		DXCoilData & Coil = DXCoil( DXCoilNum );
+
+		Coil.Name = "DX Single Speed Heating Coil";
+		Coil.DXCoilType = "Coil:Heating:DX:SingleSpeed";
+		Coil.DXCoilType_Num = CoilDX_HeatingEmpirical;
+		Coil.SchedPtr = DataGlobals::ScheduleAlwaysOn;
+		Coil.RatedSHR( 1 ) = 1.0;
+		Coil.RatedTotCap( 1 ) = 1600.0;
+		Coil.RatedCOP( 1 ) = 4.0;
+		Coil.RatedEIR( 1 ) = 1 / Coil.RatedCOP( 1 );
+		Coil.RatedAirVolFlowRate( 1 ) = 0.50;
+		Coil.RatedAirMassFlowRate( 1 ) = Coil.RatedAirVolFlowRate( 1 ) * PsyRhoAirFnPbTdbW( EnergyPlus::DataEnvironment::StdBaroPress, 21.11, 0.00881, "InitDXCoil" );
+		Coil.FanPowerPerEvapAirFlowRate( 1 ) = 773.3;
+		Coil.MinOATCompressor = -10.0;
+		Coil.CrankcaseHeaterCapacity = 0.0;
+		Coil.MaxOATDefrost = 0.0;
+		Coil.DefrostStrategy = Resistive;
+		Coil.DefrostControl = 0; // timed defrost control type
+		Coil.DefrostTime = 0.058333;
+		Coil.DefrostCapacity = 1000;
+		Coil.PLRImpact = false;
+		Coil.FuelType = FuelTypeElectricity;
+		Coil.RegionNum = 4;
+		Coil.OATempCompressorOn = -5.0;
+		Coil.OATempCompressorOnOffBlank = "true";
+		NumCurves = 5;
+		PerfCurve.allocate( NumCurves );
+		PerfomanceCurveData * pCurve;
```

Comment:
This is OK here.  This pointer isn't being used to store anything, it's just a reference.  For "reference", you don't need to do this complexity.  Just create an actual reference:

```
auto & thisCurve = PerfCurve( nCapfT );
thisCurve.CurveType = Cubic;
```

Then you don't have the sometimes yellow/red flag of a pointer variable during review.  The only downside I can think of is that you have to create multiple variable names below, one for each curve.  Unless you put them inside individual scopes (braces).

Anyway, I see your intent here, and it's OK for here, but in the future, consider the alternative approach.

Reply:
@Myoldmopar  Thanks. Comment noted.